module/execution/registers_async.go

```diff
@@ -0,0 +1,65 @@
+package execution
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/state_synchronization"
+	"github.com/onflow/flow-go/storage"
+	"go.uber.org/atomic"
+)
+
+// RegistersAsyncStore has the same basic structure as access/backend.ScriptExecutor
+// TODO: use this implementation in the `scripts.ScriptExecutor` passed into the AccessAPI
+type RegistersAsyncStore struct {
+	registerIndex storage.RegisterIndex
+	reporter      state_synchronization.IndexReporter
+	initialized   *atomic.Bool
```

Comment:
I don't think you need an atomic value here, since the only one setting the value is `InitDataAvailable` and that is done inside the `Once.Do`. Also no one sets the value back to false from what I can see.

Reply:
it's atomic so that `isDataAvailable` is thread safe